groovy/gradle/src/org/netbeans/modules/gradle/api/GradleBaseProject.java

```diff
@@ -0,0 +1,347 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.gradle.api;
+
+import org.netbeans.modules.gradle.api.execute.RunUtils;
+import java.io.File;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.netbeans.api.project.Project;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+public final class GradleBaseProject implements Serializable, ModuleSearchSupport {
```

Comment:
A project that doesn't implement `Project` interface. Why?

Reply:
Gradle side mirror object. Documentation will be provided.
Basically: NbGradleProjectImpl (the Project interface implementation) has a GradleProject (information retrieved from Gradle) has a GradleBaseProject